### DISCUSSION TITLE
Cannot find type definition file for 'cron'

### DISCUSSION THREAD

**Discussion by duynguyeenx96** [kelektiv/node-cron#717] [Q&A] [answered]
I'm having trouble running my nestjs application when I get this error.
**error TS2688: Cannot find type definition file for 'cron'.
   The file is in the program because:
     Entry point for implicit type library 'cron'**
I installed cron using npm install cron but still get this error.
I tried more
**"typeRoots": [
   "./node_modules/@types"
]**
Go to the tsconfig.json file but still get the above error.
Pls help me. Thanks

**Comment by sheerlox** [comment]
Hello, please provide us with more information about the installed packages and their versions.
Which version of `cron` are you using? Are you trying to use it with `@types/cron`?

**Comment by duynguyeenx96** [comment]
I have* cron 3.1.1* installed
I tried installing ***@***.***/cron 2.4.0* and got the message: "This is a stub
types definition. cron provides its own type definitions, so you do not
need this installed"
Despite installing *cron *or ***@***.***/cron* I still get the error
*error TS2688: Cannot find type definition file for 'cron'.*
*   The file is in the program because:*
*     Entry point for implicit type library 'cron' when running nestjs
application*
I tried deleting the node_modules and package-lock.json folders and running
the npm install command again but still no success.
This error occurs on Windows OS. Not on macOS.


Vào CN, 15 thg 10, 2023 vào lúc 19:21 Pierre Cavin <
***@***.***> đã viết:

**Reply by sheerlox** [reply]
You need to remove `@types/cron`, we shop types with the package since `2.4.0`.

**Comment by duynguyeenx96** [comment]
Yes of course I did that.
I tried upgrading typescript to the latest version but could not fix this
error.
I think there's something wrong with my computer's typescript compiler.
I tried searching for this error but it seems no one has encountered it

Vào CN, 15 thg 10, 2023 vào lúc 23:46 Pierre Cavin <
***@***.***> đã viết:

**Comment by duynguyeenx96** [comment]
[image: image.png][image: image.png]
You can see the error message in line 1 of the tsconfig.json file

Vào CN, 15 thg 10, 2023 vào lúc 23:51 Nguyễn Ngọc Duy <
***@***.***> đã viết:

**Reply by intcreator** [reply]
that image isn't showing up for me. if you're using email to reply to discussions some text is being lost in the process. try looking at the conversation directly on GitHub.com. also if you use plain text instead of screenshots your error messages will be easier to search and more useful for future visitors

**Comment by sheerlox** [comment, marked as answer]
You should either upgrade `cron` to `>=2.4.4` or install `@types/cron` pinned to `2.0.1` (which is the latest version before it was replaced by a stub). We recommend upgrading `cron` since we started shipping the types with the package in `2.4.0`! 
If you are using `cron@>=2.4.4`, remember to remove `@types/cron` from your dev dependencies 😉
